### PR TITLE
Handling footer credits for Woo Express plans. 

### PR DESCRIPTION
**Pull request by gmovr**
### Changes proposed in this Pull Request:

- We'd like to show a specific footer credit for Woo Express sites instead of the default "Blog with" for WordPress.com
- This is the Martech request: https://martechp2.wordpress.com/2023/03/10/add-footer-link-in-ecommerce-plan-sites/
- We first opened a PR in wpcomsh, but were adviced to contain this Woo Express specific code in this repo from @xristos3490 :) 

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

**Setup**
- If you haven't done this already, do an `npm i && npm run build`
- Make a free site, and tag it as a WoA test site in MC.

- Set up a Woo Express site, either a trial, or one of the paid plans via the SA.
- Checkout this branch
- Transfer all files (I excluded node_modules) to your test site via:
```sh
rsync -avze ssh --delete ./ {{my_woo_test_site}}.wordpress.com@sftp.wp.com:/{{absolute_path_to_htdocs}}/wp-content/mu-plugins/wpcomsh/vendor/automattic/wc-calypso-bridge/
```

... or probably you have a better way to sync. :) 

**Testing**
- Open another terminal window, or Tmux, and ssh into your site. Tail the error log (e.g. `less +F /tmp/php-errors`)
- When reloading your site, you should see that the footer credit is updated to
```html
<div class="wp-block-group"><!-- wp:paragraph {"align":"center"} --><p class="has-text-align-center">Powered by <a href="https://woocommerce.com/express/?utm_source=referral&amp;utm_medium=footer-credit&amp;utm_campaign=woo-express-footer-credit" rel="nofollow" class="customize-unpreviewable">Woo</a></p><!-- /wp:paragraph --></div>
```
- Verify no related PHP errors.

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: includes/class-wc-calypso-bridge-footer-credits.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    /**
+     * Get footer credit as HTML.
+     *
+     * @see <wpcomsh/block-theme-footer-credits/class-wpcom-block-theme-footer-credits.php>
+     *
+     * @return string
+     */
+    public function get_footer_credits(): string {
+        $utm_string = '?utm_source=referral&utm_medium=footer-credit&utm_campaign=woo-express-footer-credit';
+        return sprintf( 'Powered by <a href="https://woocommerce.com/express/%1$s" rel="nofollow">Woo</a>', $utm_string );
```
Comment (ilyasfoo) [inline includes/class-wc-calypso-bridge-footer-credits.php:51]: Should we translate `Powered by` text?

**Review by ilyasfoo — CHANGES_REQUESTED** [review]
Tests well, @gmovr! Just some minor comments:

- Please add a changelog update in `readme.txt` in `unreleased` section, ([example](https://github.com/Automattic/wc-calypso-bridge/pull/1261/files#diff-9e6e4772050998a5c0dc3c61acf3dab0a7e594566171fa5746d6b62f9598efb6R27)). You can create the section if it doesn't exist.
- Comment on translation:

File: includes/class-wc-calypso-bridge-footer-credits.php (hunk 1 above)
Comment (raja651) [reply includes/class-wc-calypso-bridge-footer-credits.php]: Added translation and updated changelog in [4121fa3](https://github.com/Automattic/wc-calypso-bridge/pull/1265/commits/4121fa39e6db98a34d23cfd410f25b490ac8f799)

File: includes/class-wc-calypso-bridge-footer-credits.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    /**
+     * Get footer credit as HTML.
+     *
+     * @see <wpcomsh/block-theme-footer-credits/class-wpcom-block-theme-footer-credits.php>
+     *
+     * @return string
+     */
+    public function get_footer_credits(): string {
+        $utm_string = '?utm_source=referral&utm_medium=footer-credit&utm_campaign=woo-express-footer-credit';
+        return sprintf( __( 'Powered by <a href="https://woocommerce.com/express/%1$s" rel="nofollow">Woo</a>' ), $utm_string );
```
Comment (beaulebens) [inline includes/class-wc-calypso-bridge-footer-credits.php:51]: This formatting makes things pretty hard to translate. Typically we'd try to do something like `__( 'Powered by %1$s' )` and then `sprintf` in all of the pieces (vs just the utm_string). That way the translator can't change those details (on purpose or by mistake) and just focuses on the "Powered by <something>" concept. We also wouldn't actually want them to change "Woo" here.

Something like (untested)
```
/* translators: %1$s is replaced with a link to Woo.com */
sprintf( __( 'Powered by %1$s' ), '<a href="https://woocommerce.com/express/' . $utm_string . '" rel="nofollow">Woo</a>' );
```

Also good to include the comment for translators in the line immediately before the string.

Comment (ilyasfoo) [reply includes/class-wc-calypso-bridge-footer-credits.php]: Good catch, @beaulebens! Updated in https://github.com/Automattic/wc-calypso-bridge/pull/1265/commits/23c195419e7b3a4fe93f8b827b245532ff300a0a

**Review by ilyasfoo — APPROVED** [review]
I made some fixes, tested, and I think it's safe to merge

**Pull request merged**